internal/resource/common/sourcetypeidkey/source_type_id_key_schema.go

```diff
@@ -0,0 +1,27 @@
+package sourcetypeidkey
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+func SourceTypeIdKeySchema() schema.SingleNestedAttribute {
```

Comment:
Since we have these methods all separated out into different packages, we could shorten these method names down for all these common functions (see "simplify function names" here https://go.dev/blog/package-names). For example this could just be Schema(), and it would be called like sourcetypeidkey.Schema().